lightning/src/ln/channel.rs

```diff
@@ -691,11 +691,17 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			return Err(ChannelError::Close("Insufficient funding amount for initial commitment".to_owned()));
 		}
 
+		// TODO duplicate code new_from_req / accept_channel
+		if !their_features.requires_shutdown_anysegwit() {
```

Comment:
I think this check makes it mandatory for our peers to support `opt_shutdown_anysegwit` ? I would keep it optional for now, we don't higher versions of segwit deployed yet.

Reply:
Hmmmm. That's indeed a doubt I had while I was working on this. In @TheBlueMatt's comment [here](https://github.com/rust-bitcoin/rust-lightning/issues/780#issuecomment-771896171):

> (b) we should require the feature for all peers

I assumed that he meant that this would be mandatory. Happy to make it optional if needed,

Reply:
Right, sorry, as I'd mentioned on slack the other day, I think it may be a bit too early for this. Likely we can do it in a few months, at least once a few other implementations get it shipped, but its likely preferrable to have (b) and (c) sitting in a yet-unmerged PR for now.

Reply:
Ah sorry, I missed the bit where you said it was too early ☺️. I'll rework the PR and let you know when it's ready.